core/src/scene/sceneLoader.cpp

```diff
@@ -1228,15 +1228,30 @@ void SceneLoader::loadLayers(Node layers, Scene& scene, TileManager& tileManager
         std::string name = layer.first.as<std::string>();
 
         Node data = layer.second["data"];
-        Node data_layer = data["layer"];
-        Node data_source = data["source"];
 
-        std::string collection = data_layer ? data_layer.as<std::string>() : name;
-        std::string source = data_source ? data_source.as<std::string>() : "";
+        std::string source;
+        if (Node data_source = data["source"]) {
+            if (data_source.IsScalar()) {
+                source = data_source.Scalar();
```

Comment:
Warnings would be good in many places while loading the style sheet, I could add 3 to 5 different warnings in this block alone, so before peppering log messages everywhere I'd like to think about a more general validation step before reading configuration at all.